### ISSUE TITLE
[FEATURE] Make multi-touch events possible

### ISSUE THREAD

**Issue by mmakaay** [mmakaay/esphome-xiaomi_bslamp2#14]
**Is your feature request related to a problem? Please describe.**
It currently is only possible to detect single-touch events using the binary_sensor implementation. The front panel event system provides enough information to handle multi-touch events as well. It would be really nice to expose these.

The original firmware uses a multi-touch option too. Holding the power and color buttons for a while will reset the device. The original device being able to detect multi-touch events is another good reason to implement this ;-)

**Describe the solution you'd like**

Currently, the `part` property can be used to define what part of the front panel to react to.
I propose adding a `parts` property that does the same thing, but that accepts multiple part names.
This would look like:

```yaml
binary_sensor:
  - platform: xiaomi_bslamp2
    parts:
      - power button
      - color_button
```

**Comment by mmakaay** [comment]
When handling multi-touch events, care has to be taken to get some interlocking behavior between the binary sensors that are in use. Below are some personal scribbles on how to handle the event interlocking.

| Power | Color | Slider | triggered event       | in action | remarks                 |
  |:-----:|:-----:|:------:|-----------------------|-----------|-------------------------|
  | 1     |       |        | touch P		         | P         |                         |
  | 0     |       |        | release P             |           |                         |
  |       | 1     |        | touch C               | C         |                         |
  |       | 0     |        | release C             |           |                         |
  |       |       | 1      | touch S               | S         |                         |
  |       |       | 0      | release S             |           |                         |
  | 1     |       |        | touch P               | P         |                         |
  | 1     | 1     |        | touch P+C             | P, C      |                         |
  | 1     | 0     |        | release P+C           | P, C      | no "release C" event    |
  | 0     |       |        | -                     | P, C      | no "release P" event    |
  |       | 1     |        | touch C               | C         |                         |
  | 1     | 1     |        | touch P+C             | P, C      |                         |
  | 0     | 1     |        | release P+C           | P, C      | no "release P" event    |
  | 1     | 1     |        | touch P+C             | P, C      | no "touch P" event      |
  | 1     | 0     |        | release P+C           | P, C      | no "release C" event    |
  | 0     |       |        | -                     | P, C      | no "release P" event    |

**Comment by mmakaay** [comment]
I have tried to come up with an easy to use scheme for multi-touch events, but things quickly became ugly when trying to do that within the boundaries of individual binary sensors. Therefore, I dropped the code that I wrote for this.

On a next try for multi-touch support, it probably would make most sense to implement a whole new component, into which some binary sensors can be plugged, and which handles all touch events for those sensors. That way, it is a lot cleaner to for example implement the required interlocking. Additionally, such component would work for any set of binary sensors, making it more generally usable.

For now, I'll close this issue, and will reopen it when attempt nr. 2 starts.
